### PR TITLE
Modified SmokeTest/RmCoreSmokeTests.ps1.

### PR DESCRIPTION
**Pull request by LucasYao93**
## Description

Modified SmokeTest/RmCoreSmokeTests.ps1.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by forMurphy** [comment]
@LucasYao93 Have you ran the smoke test locally? I think it's unnecessary to convert {Verb}-AzureRm{Noun} to {Verb}-Az{Noun} currently. Please revert these changes.

**Comment by BethanyZhou** [comment]
@LucasYao93 It's suggested to run the smoke test script locally. And I think it's unnecessary to convert {Verb}-AzureRm{Noun} to {Verb}-Az{Noun} currently, could you revert these changes?

**Comment by LucasYao93** [comment]
> It's suggested to run the smoke test script locally. And I think it's unnecessary to convert {Verb}-AzureRm{Noun} to {Verb}-Az{Noun} currently, could you revert these changes?

Hi @BethanyZhou ,
 I have ran the smoke test script locally which passed. Dingmeng suggested convert {Verb}-AzureRm{Noun} to {Verb}-Az{Noun}, Because the command format should be  {Verb}-Az{Noun}.

Thanks

File: tools/Test/SmokeTest/RmCoreSmokeTests.ps1 [hunk 1]
```diff
@@ -37,83 +37,101 @@ $resourceGroupName = "azpssmokerg$randomValue"
...
+                                                              $flag = $true
+                                                              Start-Sleep -Seconds 30
+                                                              if ($loopLimit -gt 30)
+                                                              {
+                                                                throw $_.Exception
+                                                              }
+                                                            }
+                                                          } while ($flag -and ($loopLimit -le 30))
+                                                        }},
+    @{Name = "Az.Resources [Cleanup]";        Command = {Remove-AzResourceGroup -Name $resourceGroupName -Force -ErrorAction Stop}}
```
Comment (BethanyZhou) [inline tools/Test/SmokeTest/RmCoreSmokeTests.ps1:62]: Please add similar retry-logic for Az.Resources [Cleanup], thanks

File: tools/Test/SmokeTest/RmCoreSmokeTests.ps1 [hunk 2]
```diff
@@ -37,83 +37,101 @@ $resourceGroupName = "azpssmokerg$randomValue"
...
+                                                            catch {
+                                                              $flag = $true
+                                                              Start-Sleep -Seconds 30
+                                                              if ($loopLimit -gt 30)
+                                                              {
+                                                                throw $_.Exception
+                                                              }
+                                                            }
+                                                          } while ($flag -and ($loopLimit -le 30))
+                                                        }},
```
Comment (BethanyZhou) [inline tools/Test/SmokeTest/RmCoreSmokeTests.ps1:61]:
```suggestion
    @{Name = "Az.Storage [Cleanup]";    
           Command = {
           $loopLimit = 0     
           do {
              try {
                 $loopLimit ++
                 Remove-AzStorageAccount -Name $storageAccountName -ResourceGroupName $resourceGroupName -Force -ErrorAction Stop
                 break
              }
              catch {
                 Start-Sleep -Seconds 30
                 Write-Warning "Retry Az.Storage [Cleanup]"
                 if ($loopLimit -gt 30)
                 {
                    Write-Warning "Give up Az.Storage [Cleanup]"
                 }
              }
           } while ($loopLimit -le 30)
        }},
```

**Review by BethanyZhou — CHANGES_REQUESTED** [review]
Enable-AzureRmAlias handles {Verb}-AzureRm{Noun} as {Verb}-Az{Noun}, I'm okay to accept these conversions.

File: tools/Test/SmokeTest/RmCoreSmokeTests.ps1 [hunk 3]
```diff
@@ -12,7 +12,7 @@ param(
 
 if (Get-Module -ListAVailable 'Az.Accounts')
 {
-    Enable-AzureRmAlias
```
[the remark this answers is not in the document]
Comment (LucasYao93) [reply tools/Test/SmokeTest/RmCoreSmokeTests.ps1]: Done

File: tools/Test/SmokeTest/RmCoreSmokeTests.ps1 (hunk 1 above)
Comment (LucasYao93) [reply tools/Test/SmokeTest/RmCoreSmokeTests.ps1]: Done

File: tools/Test/SmokeTest/RmCoreSmokeTests.ps1 (hunk 2 above)
Comment (LucasYao93) [reply tools/Test/SmokeTest/RmCoreSmokeTests.ps1]: Done
Thanks for your suggestion.

File: tools/Test/SmokeTest/RmCoreSmokeTests.ps1 [hunk 4]
```diff
@@ -37,83 +37,119 @@ $resourceGroupName = "azpssmokerg$randomValue"
...
-    @{Name = "Az.DataLakeStore";              Command = {Get-AzureRmDataLakeStoreAccount -ErrorAction Stop}},
+    @{Name = "Az.DataFactory [V1]";           Command = {Get-AzDataFactory -ResourceGroupName $resourceGroupName -ErrorAction Stop}},
+    @{Name = "Az.DataFactoryV2 [V2]";         Command = {Get-AzDataFactoryV2 -ErrorAction Stop}},
+    @{Name = "Az.DataLakeAnalytics";          Command = {Get-AzDataLakeAnalyticsAccount -ErrorAction Stop}},
+    @{Name = "Az.DataLakeStore";              Command = {Get-AzDataLakeStoreAccount -ErrorAction Stop}},
     @{Name = "Az.DataShare";                  Command = {Get-AzDataShareAccount -ResourceGroupName $resourceGroupName -ErrorAction Stop}},
     # Waiting for an issue fix: https://github.com/Azure/azure-powershell/issues/13522#issuecomment-728659457
     # @{Name = "Az.DeploymentManager";          Command = {try {Get-AzDeploymentManagerArtifactSource -ResourceGroupName $resourceGroupName  -ErrorAction Stop}catch {if ($_.ToString() -notlike "*not found*") {throw $_}}}},
     # Waiting for api '2020-11-02-preview' ready
```
Comment (BethanyZhou) [inline tools/Test/SmokeTest/RmCoreSmokeTests.ps1:109]: Please remove this comment

**Review by BethanyZhou — CHANGES_REQUESTED** [review]
Leave one comment. Others LGTM

File: tools/Test/SmokeTest/RmCoreSmokeTests.ps1 (hunk 4 above)
Comment (LucasYao93) [reply tools/Test/SmokeTest/RmCoreSmokeTests.ps1]: Done

**Review by BethanyZhou — DISMISSED** [review]

File: tools/Test/SmokeTest/RmCoreSmokeTests.ps1 [hunk 5]
```diff
@@ -37,83 +37,118 @@ $resourceGroupName = "azpssmokerg$randomValue"
...
+                                                                    {
+                                                                        Write-Warning "Give up Az.Resources [Cleanup]"
+                                                                        break
+                                                                    } else {
+                                                                        Start-Sleep -Seconds 30
+                                                                        Write-Warning "Retry Az.Resources [Cleanup]"
+                                                                    }
+                                                                }
+                                                            } while ($true)
+                                                        }}
```
Comment (dingmeng-xue) [inline tools/Test/SmokeTest/RmCoreSmokeTests.ps1:80]:
```suggestion
    @{Name = "Az.Resources [Cleanup]";        Command = {
                                                            $loopLimit = 0
                                                            do {
                                                                try {
                                                                    Remove-AzResourceGroup -Name $resourceGroupName -Force -ErrorAction Stop
                                                                    break
                                                                }
                                                                catch {
                                                                    if (++$loopLimit -gt 30)
                                                                    {
                                                                        Write-Warning "Give up Az.Resources [Cleanup]"
                                                                        break
                                                                    } else {
                                                                        Start-Sleep -Seconds 30
                                                                        Write-Warning "Retry Az.Resources [Cleanup]"
                                                                    }
                                                                }
                                                            } while ($true)
                                                        }}
```
Remove resource group has no similar problem. The code doesn't need to be changed.

Comment (LucasYao93) [reply tools/Test/SmokeTest/RmCoreSmokeTests.ps1]: Done

File: tools/Test/SmokeTest/RmCoreSmokeTests.ps1 [hunk 6]
```diff
@@ -37,83 +37,100 @@ $resourceGroupName = "azpssmokerg$randomValue"
...
+                                                            do {
+                                                                try {
+                                                                    Remove-AzStorageAccount -Name $storageAccountName -ResourceGroupName $resourceGroupName -Force -ErrorAction Stop
+                                                                    break
+                                                                }
+                                                                catch {
+                                                                    if (++$loopLimit -gt 30)
+                                                                    {
+                                                                        throw "Failed to invoke Az.Storage [Cleanup]. $_"
+                                                                        break
```
Comment (BethanyZhou) [inline tools/Test/SmokeTest/RmCoreSmokeTests.ps1:54]: this break can be removed

**Review by BethanyZhou — CHANGES_REQUESTED** [review]
LGMT. Leaved one comment.

File: tools/Test/SmokeTest/RmCoreSmokeTests.ps1 (hunk 6 above)
Comment (LucasYao93) [reply tools/Test/SmokeTest/RmCoreSmokeTests.ps1]: Done

**Review by BethanyZhou — APPROVED** [review]

**Review by BethanyZhou — APPROVED** [review]

**Pull request merged**